src/sortedarray.c

```diff
@@ -0,0 +1,244 @@
+/*
+
+Copyright (c) 2016, Stefan Cloudt
+
+Permission to use, copy, modify, and/or distribute this software
+for any purpose with or without fee is hereby granted, provided
+that the above copyright notice and this permission notice appear
+in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ 
+ */
+
+/**
+ * @file sortedarray.c
+ * 
+ * @brief File containing the implementation of sortedarray.h
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "sortedarray.h"
+
+#ifdef ALLOC_TESTING
+#include "alloc-testing.h"
+#endif
+
+/**
+ * Find the first index where the cmp_func of the sortedarray returns 0. There
+ * must be an element in the range from b to e where cmp_func returns 0 to get
+ * expected results.
+ *
+ * @param sortedarray   The SortedArray.
+ * @param data          The value to compare to.
+ * @param b             The start of the range (inclusive).
+ * @param e             The end of the range (inclusive).
+ *
+ * @return The first index where cmp_func returns 0.
+ */
+static unsigned int sortedarray_first_index(SortedArray *sortedarray,
```

Comment:
Avoid function prototypes where possible: move these functions up to the top of the file. It's not necessary to attach full doc comments for internal (static) functions.